### ISSUE TITLE
Release process

### ISSUE THREAD

**Issue by f4hy** [fsspec/filesystem_spec#1200]
I was expecting there to be a February release but looks like there wasn't one. I am curious about the release cadence of fsspec, I don't see a document in the repo outlining the release process. 

There are number of fixes that have been merged but yet to be released that I am waiting on. It would be nice if there was some guidance on expectation of how often releases are. Thanks

**Comment by martindurant** [comment]
You are right that we could be clearer, but we generally release roughly once a month. There will be one coming soon, as maintainer time allows.

**Comment by f4hy** [comment]
Thanks for the update. If there is anything I can help with happy to draft or review a release info document.

**Comment by efiop** [comment]
We might consider doing nightly builds automatically (they will be marked as pre-releases not to interefere with proper releases). But that depends on https://github.com/fsspec/community/issues/6 to formalize release process, as current manual normal releases are clearly a no-go and don't scale at all putting more pressure on @martindurant specifically. Been meaning to do that myself, but can't find time for it for now 🙁 hopefully soon.

**Comment by martindurant** [comment]
There are a couple of complicating factors:
- we do not maintain the release notes until release time. We could auto-generate this if we were careful about merge messages, or require PR submitters to add an entry before we consider a merge. I don't think the current log is very useful
- fsspec/s3fs/gcsfs are always released together with the same version tag, and in that order (even if one of them has no changes). adlfs often follows suit.

**Comment by martindurant** [comment]
fsspec and s3fs 2023.3.0 are on pypi
